src/domain/authentication/__tests__/UnauthorizedPage.test.jsx

```diff
@@ -0,0 +1,124 @@
+
+import UnauthorizedPage from '../UnauthorizedPage';
+
+// Mock the dependencies
+vi.mock('react-admin', async (importOriginal) => {
+  const actual = await importOriginal();
+  return {
+    ...actual,
+    useLogout: vi.fn(),
+    useTranslate: vi.fn(),
+  };
+});
+
+// Mock mailto link behavior
+vi.mock('react-router-dom', async () => {
+  const actual = await vi.importActual('react-router-dom');
+  return {
+    ...actual,
+    useNavigate: () => vi.fn(),
+  };
+});
+
+// Mock InfoPageTemplate
+vi.mock('./InfoPageTemplate', () => ({
+  __esModule: true,
+  default: ({ title, content }) => (
+    <div data-testid="info-page-template">
+      <div data-testid="info-page-title">{title}</div>
+      <div data-testid="info-page-content">{content}</div>
+    </div>
+  ),
+}));
+
+describe('UnauthorizedPage', () => {
+  const queryClient = new QueryClient();
+
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('renders the UnauthorizedPage with correct content', () => {
+    // Arrange
+    const mockTranslate = vi.fn((key) => {
+      switch (key) {
+        case 'authentication.unauthorizedPage.title':
+          return 'Unauthorized Title';
+        case 'authentication.unauthorizedPage.content':
+          return 'Unauthorized Content';
+        case 'authentication.unauthorizedPage.contactEmail':
+          return 'test@example.com';
+        case 'ra.auth.logout':
+          return 'Logout';
+        default:
+          return key;
+      }
+    });
+    useTranslate.mockReturnValue(mockTranslate);
+    useLogout.mockReturnValue(vi.fn());
+
+    // Act
+    render(
+      <QueryClientProvider client={queryClient}>
+        <MemoryRouter>
+          <UnauthorizedPage />
+        </MemoryRouter>
+      </QueryClientProvider>
+    );
+
+    // Assert
+    expect(screen.getByText('Unauthorized Title')).toBeInTheDocument();
+    expect(screen.getByText('Unauthorized Content')).toBeInTheDocument();
+
+    const link = screen.getByRole('link');
+    expect(link).toHaveAttribute('href', 'mailto:test@example.com');
+    expect(link).toHaveTextContent('test@example.com');
+
+    expect(screen.getByRole('button')).toBeInTheDocument();
+    expect(screen.getByRole('button')).toHaveTextContent('Logout');
+  });
+
+  it('calls logout when the logout button is clicked', async () => {
+    // Arrange
+    const mockLogout = vi.fn().mockResolvedValue(undefined);
+    useLogout.mockReturnValue(mockLogout);
+    useTranslate.mockReturnValue(
+      vi.fn((key) => {
+        switch (key) {
+          case 'authentication.unauthorizedPage.title':
+            return 'Unauthorized Title';
+          case 'authentication.unauthorizedPage.content':
+            return 'Unauthorized Content';
+          case 'authentication.unauthorizedPage.contactEmail':
+            return 'test@example.com';
+          case 'ra.auth.logout':
+            return 'Logout';
+          default:
+            return key;
+        }
+      })
+    );
```

Comment:
This could be moved to its own function, or possibly set as the useTranslate's mock in one place, to reduce code duplication in this file.